util/EIP-2124/src/lib.rs

```diff
@@ -95,77 +96,126 @@ pub struct ForkId {
 pub enum RejectReason {
 	/// Remote node is outdated and needs a software update.
 	RemoteStale,
-	/// Local node is on an incompatible chain or needs a sofwtare update.
+	/// Local node is on an incompatible chain or needs a software update.
 	LocalIncompatibleOrStale,
 }
 
 /// Filter that describes the state of blockchain and can be used to check incoming `ForkId`s for compatibility.
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, MallocSizeOf)]
+#[cfg_attr(any(test, feature = "test-helpers"), derive(PartialEq))]
 pub struct ForkFilter {
-	/// Blockchain head
-	pub head: BlockNumber,
-	past_forks: BTreeMap<BlockNumber, ForkHash>,
-	next_forks: BTreeSet<BlockNumber>,
+	forks: BTreeMap<BlockNumber, ForkHash>,
+
+	head: BlockNumber,
+
+	cache: Cache,
+}
+
+#[derive(Clone, Debug, MallocSizeOf)]
+#[cfg_attr(any(test, feature = "test-helpers"), derive(PartialEq))]
+struct Cache {
+	epoch_start: BlockNumber,
+	epoch_end: Option<BlockNumber>,
```

Comment:
the `epoch` is a very overloaded term, would be nice to have some internal docs here

Reply:
still would be nice to have ^^^ (it doesn't seem to refer to [ethash `epoch`](https://github.com/ethereum/wiki/wiki/Mining#so-what-is-mining-anyway), does it?)

Reply:
done in `ethereum-forkid` crate